src/scripts/background/connection/callbacks/ProtocolCallbacks.ts

```diff
@@ -17,10 +20,41 @@ async function addWebpage([url]: [string]): Promise<void> {
   await getMintingRegistrar().maybeTryRegister();
 }
 
+async function isWhitelisted() {
+  return new Promise((resolve, reject) => {
+    chrome.tabs.query(
+      { currentWindow: true, active: true },
+      function (tabArray) {
+        let currentUrl = new URL(tabArray[0].url as string);
+        resolve(WHITELISTED.includes(currentUrl.host));
+      },
+    );
+  });
+}
+
+async function getTotalFactsCount(): Promise<number> {
+  if (!(await isWhitelisted())) {
+    return 0;
+  }
+  return getDataHost().length();
+}
+
+async function getFact(index: number): Promise<IFact> {
+  if (!(await isWhitelisted())) {
+    return Promise.resolve({} as IFact);
+  }
```

Comment:
Please confirm if this check happens in the extension's context or the webpage's.

Reply:
I do confirm.